website/docs/docs/build/custom-schemas.md

```diff
@@ -192,12 +157,51 @@ See docs on macro `dispatch`: ["Managing different global overrides across packa
 
 </VersionBlock>
 
+## A built-in alternative pattern for generating schema names
+
+A common customization is to ignore the target schema in production environments, and ignore the custom schema configurations in other environments (such as development and CI).
+
+Production Environment (`target.name == 'prod'`)
+
+| Target schema | Custom schema | Resulting schema |
+| ------------- | ------------- | ---------------- |
+| analytics_prod | None | analytics_prod |
+| analytics_prod | marketing | marketing |
+
+Development/CI Environment (`target.name != 'prod'`)
+
+| Target schema | Custom schema | Resulting schema |
+| ------------- | ------------- | ---------------- |
+| alice_dev | None | alice_dev |
+| alice_dev | marketing | alice_dev |
+| dbt_cloud_pr_123_456 | None | dbt_cloud_pr_123_456 |
+| dbt_cloud_pr_123_456 | marketing | dbt_cloud_pr_123_456 |
+
+Just like the normal macro, this approach guarantees that schemas from different environments will not collide.
+
+dbt ships with a macro for this use case – called `generate_schema_name_for_env` – which is disabled by default. To enable it, add a custom `generate_schema_name` macro to your project that contains the following code:
+
+<File name='macros/get_custom_schema.sql'>
+
+```sql
+-- put this in macros/get_custom_schema.sql
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+    {{ generate_schema_name_for_env(custom_schema_name, node) }}
+{%- endmacro %}
+```
+
+</File>
+
+**Note:** When using this macro, you'll need to set the target name in your production job to `prod`.
+
 ## Managing environments
 
-In the `generate_schema_name` macro examples shown above, the `target.name` context variable is used to change the schema name that dbt generates for models. If the `generate_schema_name` macro in your project uses the `target.name` context variable, you must additionally ensure that your different dbt environments are configured appropriately. While you can use any naming scheme you'd like, we typically recommend:
- - **dev**: Your local development environment; configured in a `profiles.yml` file on your computer.
+In the `generate_schema_name` macro examples shown above, the `target.name` context variable is used to change the schema name that dbt generates for models. If the `generate_schema_name` macro in your project uses the `target.name` context variable, you must ensure that your different dbt environments are configured accordingly. While you can use any naming scheme you'd like, we typically recommend:
```

Comment:
```suggestion
In the `generate_schema_name` macro examples shown in the section [](/docs/build/custom-schemas#a-built-in-alternative-pattern-for-generating-schema-names), the `target.name` context variable is used to change the schema name that dbt generates for models. If the `generate_schema_name` macro in your project uses the `target.name` context variable, you must ensure that your different dbt environments are configured accordingly. While you can use any naming scheme you'd like, dbt Labs typically recommends:
```

Reply:
Handled this one separately as your suggestion didn't have anchor text, but it's resolved now